parlai/crowdsourcing/tasks/qa_data_collection/run.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import os
+import sys
+from mephisto.operations.operator import Operator
+from mephisto.tools.scripts import load_db_and_process_config
+from mephisto.abstractions.blueprints.parlai_chat.parlai_chat_blueprint import (
+    BLUEPRINT_TYPE,
+    SharedParlAITaskState,
+)
+
+import hydra
+from omegaconf import DictConfig, MISSING
+from dataclasses import dataclass, field
+from typing import List, Any
+
+from parlai.tasks.squad.agents import DefaultTeacher
+
+# This needs to be properly streamlined
+class SquadDataLoader(DefaultTeacher):
```

Comment:
I think you can move this to a `util.py` module to keep `run.py` relatively uncluttered - also, the more ParlAI-standard way to write such a wrapper teacher would be to subclass `parlai.tasks.wrapper.agents.AbstractWrapperTeacher` and then define a custom `._edit_action()` method in which to modify the `'text'` field of the action

Reply:
Does `AbstractWrapperTeacher` and `parlai.core.worlds.create_task` have different workflows? For SQuAD, `AbstractWrapperTeacher` was asking for `datapath`, but `create_task` is working without `datapath`

Reply:
Oh hmm, you could actually combine `AbstractWrapperTeacher` and `create_task` together, like this:
```
from parlai.agents.repeat_label.repeat_label import RepeatLabelAgent
from parlai.core.params import ParlaiParser
from parlai.core.worlds import create_task


args = """\
--task wrapper:LabelToText
--wrapper-task squad
--datatype train
"""
parser = ParlaiParser(True, False)
opt = parser.parse_args(args.split())
agent = RepeatLabelAgent(opt)
teacher = create_task(opt, agent).get_task_agent()
```
Here, `LabelToText` is a teacher that subclasses `AbstractWrapperTeacher`, but you could make a similar subclassed teacher that does the string manipulation that you want

Reply:
Thanks! :) I was looking for something as simple as this! Made changes in 104e07a0f49587b2f37b91ab19384d4be28106b6

Reply:
Okay great, that works! But you probably don't need both a `task` and `wrapper_task` config param in `run.py` - it looks like this wrapper teacher is specific to SQuAD, so you might as well just hardcode in that `--wrapper-task` has to be `'squad'`, maybe by adding in a `parser.set_defaults(wrapper_task='squad')` line in your subclass's `.add_cmdline_args()` method. Also, since this subclassed wrapper teacher is SQuAD-specific, it's likely more suitable to go in `parlai.tasks.squad.agents` than in `parlai.tasks.wrapper.agents`

Reply:
Changes to remove `wrapper_task` are in 870690f71c9829028ddd9639532a72d281aa387c

Reply:
Ah fantastic! Hmm, in the TeacherConfig, can you set the default to `"squad:SquadQATeacher"`, just so that the teacher works on the default settings? Also, super minor comment, but in the SquadQATeacher, it might be good to put the add_cmdline_args() method before the other methods for consistency, since we typically put the classmethods before the rest. Other than that I think this looks good :)